funsor/numpy.py

```diff
@@ -7,6 +7,45 @@
 from funsor.util import quote
 
 
+def einsum(equation, *operands):
+    """
+    Log-sum-exp implementation of einsum.
+    """
+    # rename symbols to support PyTorch 0.4.1 and earlier,
```

Comment:
nit: Does this still apply to NumPy?

Reply:
It applies to NumPy, but in JAX we don't need this.